proto/options.proto

```diff
@@ -106,11 +111,11 @@ message CostingOptions {
   optional uint32 flow_mask = 55;
   optional float rail_ferry_cost = 56;
   optional float use_rail_ferry = 57;
-}
 
-message AvoidEdge {
-  optional uint64 id = 1;
-  optional float percent_along = 2;
+  // these are not specified directly by the user but they get filled in as the request is parsed and fulfilled
+  optional Costing costing = 90;
+  optional string name = 91;
+  repeated AvoidEdge avoid_edges = 92;                       // Avoid edges for any costing - derived from avoid_locations
```

Comment:
i moved this inside of the costing because in the end that is where the information ends up. this is necessary after making it so that costing is selfcontained in the costing options object